dlib/dnn/layers.h

```diff
@@ -3985,31 +4079,28 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    template <unsigned long s_mode_>
```

Comment:
```suggestion
    template <operation_mode s_mode_>
```
Change these to use `operation_mode` too though.  Like don't use an integer anywhere to represent the `operation_mode` but instead use `operation_mode` to represent the `operation_mode` :)

Reply:
Thanks for updating the code accordingly.